### PR TITLE
Feature/accumulate position estimates

### PR DESCRIPTION
**Pull request by hannahcl**
Accumulate measurements over time and fit a normal curve.  

Partially addressing this issue: [ Develop ROS node for accumulating estimation data #80 ](https://github.com/vortexntnu/acoustics/issues/80)

### CODE REVIEW

File: Python/accumulate_position_estimates/accumulate_position_estimates.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        self.m_last_elements = m_last_elem
+        self.distance_threshold = distance_threshold
+
+    def integrate_new_measurement(self, measurment: list):
+
+        if (
+            measurment[0] < pool_size.x
+            and measurment[1] < pool_size.y
+            and measurment[2] < pool_size.z
+        ):
```
Comment (nicbres) [inline Python/accumulate_position_estimates/accumulate_position_estimates.py:29]: probably better to make a function as in `check_pool_boundaries`. We can most likely throw away the negative outliers as well.

And it's `measurement` not `measurment` :)

File: Python/accumulate_position_estimates/accumulate_position_estimates.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+
+pool_size = Position(25, 12.5, 5)
+
+
+class PositionEstimate:
+    def __init__(self, m_last_elem, distance_threshold) -> None:
+        self.mean = Position(0, 0, 0)
+        self.variance = Position(0, 0, 0)
+        self.measurements = [[], [], []]
```
Comment (nicbres) [inline Python/accumulate_position_estimates/accumulate_position_estimates.py:18]: Is there a reason measurements are not a list of position objects?

You can extend that class to allow summation, subtraction, etc.

File: Python/accumulate_position_estimates/accumulate_position_estimates.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+            prev_mean_x = sum(self.measurements[0][: n - m]) / (n - m)
+            prev_mean_y = sum(self.measurements[1][: n - m]) / (n - m)
+            prev_mean_z = sum(self.measurements[2][: n - m]) / (n - m)
+
+            current_mean_x = sum(self.measurements[0][n - m :]) / m
+            current_mean_y = sum(self.measurements[1][n - m :]) / m
+            current_mean_z = sum(self.measurements[2][n - m :]) / m
+
+            if self.distance_threshold.x < abs(prev_mean_x - current_mean_x):
```
Comment (nicbres) [inline Python/accumulate_position_estimates/accumulate_position_estimates.py:65]: We could possibly think about making the threshold dependent on the current variance. But I guess this is fine for now.

**Review by nicbres — CHANGES_REQUESTED** [review]

File: Python/accumulate_position_estimates/accumulate_position_estimates.py (hunk 3 above)
Comment (hannahcl) [reply Python/accumulate_position_estimates/accumulate_position_estimates.py:65]: If we look at it as a Bayesian detector with
H0 : delta = 0, 
H1: delta = distance between pingers, where delta = mean of first n-m measurement -  mean of last m measurements, 
then we get the decision rule delta >< distance between pinger / 2. 

The variance of the two conditional likelihood functions will be the same right?

File: Python/accumulate_position_estimates/accumulate_position_estimates.py (hunk 2 above)
Comment (hannahcl) [reply Python/accumulate_position_estimates/accumulate_position_estimates.py:18]: I used a list of position objects at first, but it was a lot easier to find mean and variance of al x, y and z positions when I used three lists instead. 
Do you think it is better (why?) to use position objects and special methods?

Comment (PizzaAllTheWay) [reply Python/accumulate_position_estimates/accumulate_position_estimates.py:19]: seems legit jesjes

**Review by PizzaAllTheWay — APPROVED** [review]
Good stuff <3

**Comment by PizzaAllTheWay** [comment]
Closing due to it being old, will keep the branch, but not the pull request....

**Pull request closed without merging**